### DISCUSSION TITLE
Environment Variables -- Best Way to set them?

### DISCUSSION THREAD

**Discussion by cr2610** [thijsvanloef/palworld-server-docker#184] [Q&A]
I am utilizing the docker compose method to change server settings, but It does not seem to be affected in the PalWorldSettings.ini file. I placed DEATH_PENALTY=NONE under the environment section like the below for example:
      environment:
         - PUID=1000
         - PGID=1000
         - PORT=8211 # Optional but recommended
         - PLAYERS=16 # Optional but recommended
         - SERVER_PASSWORD="worldofpals" # Optional but recommended
         - MULTITHREADING=true
         - RCON_ENABLED=true
         - RCON_PORT=25575
         - TZ=UTC
         - ADMIN_PASSWORD="adminPasswordHere"
         - COMMUNITY=false  # Enable this if you want your server to show up in the community servers tab, USE WITH SERVER_PASSWORD!
         - SERVER_NAME="World of Pals"
         - SERVER_DESCRIPTION="Awesome World of Pal"
         - DEATH_PENALTY=NONE

Any ideas? Additionally, is there a better way to do this, so that I don't create a new container should I need to update them? the .bashrc method I wouldn't think is the best. Thanks!

**Comment by thijsvanloef** [comment]
@cr2610 Please make sure you are running the latest image. 

However I believe the settings are case sensitive and NONE would not work, could you try: `none`?

**Reply by cr2610** [reply]
@thijsvanloef Thanks I will give this a shot. Is it expected for the PalWorldSettings.ini to be populated with the settings you change through environment variables? If it's not, that's fine. I just wanted to confirm the change is taken upon restarting container. Thank you

**Reply by thijsvanloef** [reply]
Yes the ENV vars should populate the PalWorldSettings.ini

**Comment by cr2610** [comment]
@thijsvanloef Yeah it's still not being updated in the option settings parameter of PalWorldSettings.ini. I got the env list from here: https://shockbyte.com/billing/knowledgebase/1189/How-to-Configure-your-Palworld-server.html . This is my compose. I can open an issue if thats better. 

services:
   palworld:
      image: thijsvanloef/palworld-server-docker:latest
      restart: unless-stopped
      container_name: palworld-server
      ports:
        - 8211:8211/udp
        - 27015:27015/udp # Required if you want your server to show up in the community servers tab
      environment:
         - PUID=1000
         - PGID=1000
         - PORT=8211 # Optional but recommended
         - PLAYERS=16 # Optional but recommended
         - SERVER_PASSWORD="changed" # Optional but recommended
         - MULTITHREADING=true
         - RCON_ENABLED=true
         - RCON_PORT=25575
         - TZ=UTC
         - ADMIN_PASSWORD="changed"
         - COMMUNITY=false  # Enable this if you want your server to show up in the community servers tab, USE WITH SERVER_PASSWORD!
         - SERVER_NAME="changed"
         - SERVER_DESCRIPTION="changed"
         - DEATH_PENALTY=None
      volumes:
         - ./palworld:/palworld/

**Comment by zbellerose** [comment]
Alternatively you can edit the PalWorldSettings.ini manually in your mounted volume. I've done this and had no issues.
